ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -690,7 +692,10 @@ def runTest(self):
 
             if no_routing_stop:
                 self.log("Downtime was %s" % str(no_routing_stop - no_routing_start))
-                self.log("Reboot time was %s" % str(no_routing_stop - self.reboot_start))
+                if no_routing_stop != datetime.datetime.min:
+                    self.log("Reboot time was %s" % str(no_routing_stop - self.reboot_start))
+                else:
+                    self.log("Reboot time was minimal")
```

Comment:
I'd put here: Failed to record reboot time.

Reply:
I'm afraid that `Failed` keyword would confuse a watcher. Another cause to keep the message - it is possible to use some external regexp to parse CT results. In this case we will not break the regex.

Any way - the situation just show a particular datapline feature, not the issue.
Let me know if you insist on the change.

Reply:
I think it's impossible for us to measure data plane disruption time with such good granularity.
Also you're using datetime.min here as a marker of timeout while waiting of dataplane start/stop.
So I would suggest don't use this marker value for anything.

Reply:
just added one more variable indicating the fact that dataplane never stopped

Reply:
> I'd suggest to improve this PR:
> 
> 1. Remove time.sleep. We want to send packets as fast as possible
> 2. If a goal of this PR to fix the crash, let's do that: we need to define upper_replies in case of timeout exception.
>    Also I think as soon as we have timeout exception, we should fail the test

1) I have removed  time.sleep
2) I introduced additional logic (variable 'routing_always') to be able to print "Reboot time was" message correctly
3) I don't agree to fail the test in case 'timeout exception' as it just indicates the fact that the dataplane never went down. Nothing criminal from the `fast-reboot` case perspective

Reply:
1. Thanks
2. -
3. The dataplane must gone done in case of fast-reboot.

Reply:
Could you please propose a way how to detect small periods using PTF? We need this because traffic disruption in our case is ~70ms. So, dataplane goes down but PTF can't detect such small interval. The current test implementation is able to detect such small interval once per 4..5 tries (but Ixia, for example, shows disruption reliably). In this PR I've tried to set intervals less than some minimum period to "0:00:00"